docs/branch_tracking.md

```diff
@@ -0,0 +1,53 @@
+# Branch tracking
+
```

Comment:
Can we expand the "Remotes" section in `branches.md`?

I think the "Remotes" section can be reorganized as follows:
* "Jujutsu identifies a branch by its name across remotes"...
  * "any branches from the remote will be imported" *by default*
* About `branch@remote` syntax: "Jujutsu also records the last seen position"...
* About importing remote branches: "When you pull from a remote, "... *by default*
  * can be turned off by `git.auto-local-branch = false`
  * `jj branch track`/`untrack` to control the behavior per remote branch

I have a TODO item to update `branches.md`, but not started yet. ;)

Reply:
I haven't looked at the details, but I generally agree with Yuya. In particular, I think that what you wrote can go into sub-sections of the Remotes section in `branches.md`.

I think it'd be OK to not polish everything completely right in this PR (if you don't feel like it), but to simply add subsections about (Un)tracking of branches to that section.

Mentioning what is affected by the `git.auto-local-branches` option and how seems important. It might be worth making a dedicated subsection for that, unless it goes nicely in one of the other sections. OTOH, if you don't feel like doing it, you could add a stub for it and insert "by default" in a few sections.

Reply:
> Can we expand the "Remotes" section in `branches.md`?

This is done now. 

> I think it'd be OK to not polish everything completely right in this PR (if you don't feel like it), but to simply add subsections about (Un)tracking of branches to that section.

I think I can expand the doc to explain branch tracking but I'd leave the polishing part to someone else. 

> Mentioning what is affected by the `git.auto-local-branches` option and how seems important. It might be worth making a dedicated subsection for that, unless it goes nicely in one of the other sections.

I'm up to it, as long I have your guidance.

Reply:
I've added a section on `git.auto-local-branches` in 2fae523.